libsolidity/analysis/ControlFlowGraph.cpp

```diff
@@ -0,0 +1,566 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libsolidity/analysis/ControlFlowGraph.h>
+
+#include <boost/range/adaptor/reversed.hpp>
+
+#include <algorithm>
+
+using namespace std;
+using namespace dev::solidity;
+
+namespace dev
+{
+namespace solidity
+{
+
+/** Helper class that parses the control flow of a non-top-level ASTNode. */
+class ControlFlowParser: private ASTConstVisitor
```

Comment:
Since it is not a friend anymore, it might make sense to move it to its own file. Also perhaps rather call it `Builder` than `Parser`?

Reply:
OK, I can do that :).